### PR TITLE
 doc: translate hooks-state

### PR DESCRIPTION
**Pull request by piovischioh**

### CODE REVIEW

File: content/docs/hooks-state.md [hunk 1]
```diff
@@ -98,17 +99,17 @@ function Example() {
 }
 ```
 
-**What is a Hook?** A Hook is a special function that lets you "hook into" React features. For example, `useState` is a Hook that lets you add React state to function components. We'll learn other Hooks later.
+**什麼是 Hook？**Hook 是一個讓你可以使用 React 各項功能的特殊 function。舉例來說，`useState` 是一個讓你添加 React state 到 function component 的 Hook。在後面的章節我們會學習其他 Hook。 
 
-**When would I use a Hook?** If you write a function component and realize you need to add some state to it, previously you had to convert it to a class. Now you can use a Hook inside the existing function component. We're going to do that right now!
+**什麼時候該使用 Hook？**以前當你寫一個 function component 你需要增加一些 state 時，你必須轉換成 class。現在你可以直接在 function component 中使用 Hook。我們現在就來試試看吧！
```
Comment (neighborhood999) [inline content/docs/hooks-state.md]:
```suggestion
**什麼時候該使用 Hook？**以前當你寫一個 function component 需要增加一些 state 時，你必須轉換成 class。現在你可以直接在 function component 中使用 Hook。我們現在就來試試看吧！
```

File: content/docs/hooks-state.md [hunk 2]
```diff
@@ -98,17 +99,17 @@ function Example() {
...
+>注意
 >
->There are some special rules about where you can and can't use Hooks within a component. We'll learn them in [Rules of Hooks](/docs/hooks-rules.html).
+>關於在 component 中哪裡可以使用 Hook，有一些特別的規則。我們會在 [Hook 的規則](/docs/hooks-rules.html) 中學習。
 
-## Declaring a State Variable {#declaring-a-state-variable}
+## 宣告一個 State 變數 {#declaring-a-state-variable}
 
-In a class, we initialize the `count` state to `0` by setting `this.state` to `{ count: 0 }` in the constructor:
+在 class 中，藉由在 constructor 設置 `this.state` 成 `{ count: 0 }` 把 `count` 這個 state 起始值設為 `0`。
```
Comment (neighborhood999) [inline content/docs/hooks-state.md]:
```suggestion
在 class 中，藉由在 constructor 設定 `this.state` 成 `{ count: 0 }` 把 `count` 這個 state 起始值設為 `0`。
```

File: content/docs/hooks-state.md [hunk 3]
```diff
@@ -120,76 +121,76 @@ class Example extends React.Component {
...
+**呼叫 `useState` 做了什麼？**它宣告了一個「state 變數」。我們的變數叫做 `count`，也可以有其他稱呼，像是 `banana`。這是一個在 function 呼叫中「保留」變數的方法－`useState` 就像是 class 中 `this.state` 的功能一樣。一般情況下，變數會在 function 結束時「消失」，但 state 變數會被 React 保留起來。
 
-**What do we pass to `useState` as an argument?** The only argument to the `useState()` Hook is the initial state. Unlike with classes, the state doesn't have to be an object. We can keep a number or a string if that's all we need. In our example, we just want a number for how many times the user clicked, so pass `0` as initial state for our variable. (If we wanted to store two different values in state, we would call `useState()` twice.)
+**我們傳入什麼參數給 `useState`？**唯一需要傳入 `useState()` Hook 的參數就是 state 起始值。不像 class state 不需要是一個 object。我們可以使用我們所需的 number 或是 string。在範例中，我們只需要一個 number 紀錄使用者點擊了多少次，所以我們傳入 `0` 來當作起始值。（如果要在 state 保存兩個不同的值，可以呼叫 `useState()` 兩次。）
```
Comment (neighborhood999) [inline content/docs/hooks-state.md]:
```suggestion
**我們傳入什麼參數給 `useState`？**唯一需要傳入 `useState()` Hook 的參數就是 state 起始值。不像 class state 不需要是一個 object。我們可以使用我們所需的 number 或是 string。在範例中，我們只需要一個 number 記錄使用者點擊了多少次，所以我們傳入 `0` 來當作起始值。（如果要在 state 儲存兩個不同的值，可以呼叫 `useState()` 兩次。）
```

File: content/docs/hooks-state.md [hunk 4]
```diff
@@ -120,76 +121,76 @@ class Example extends React.Component {
...
+我們宣告了一個叫做 `count` 的 state 變數，並將起始值設成了 `0`。React 在 re-render 間會記住目前的值，並將它提供給我們的 function。如果我們需要更新目前的 `count`，我們可以呼叫 `setCount`。
 
->Note
+>注意
 >
->You might be wondering: why is `useState` not named `createState` instead?
+>你也許會好奇：為什麼 `useState` 不叫做 `createState`？
 >
->"Create" wouldn't be quite accurate because the state is only created the first time our component renders. During the next renders, `useState` gives us the current state. Otherwise it wouldn't be "state" at all! There's also a reason why Hook names *always* start with `use`. We'll learn why later in the [Rules of Hooks](/docs/hooks-rules.html).
+>state 只會在第一次 render 時被建立，所以「Create」並不會比較精準。而在其後的 render，`useState` 只會給我們目前的 state。不然這就不叫做 state 啦！另外還有一個理由，為什麼 Hook *總是*以 `use` 起頭。我們稍後會在 [Hook 規則](/docs/hooks-rules.html)中瞭解。
```
Comment (neighborhood999) [inline content/docs/hooks-state.md]:
```suggestion
>state 只會在第一次 render 時被建立，所以「Create」並不會比較精準。而在其後的 render，`useState` 只會給我們目前的 state。不然這就不叫做 state 啦！另外還有一個理由，為什麼 Hook *總是*以 `use` 起頭。我們稍後會在 [Hook 規則](/docs/hooks-rules.html)中了解。
```

File: content/docs/hooks-state.md [hunk 5]
```diff
@@ -120,76 +121,76 @@ class Example extends React.Component {
...
 
-## Reading State {#reading-state}
+## 讀取 State {#reading-state}
 
-When we want to display the current count in a class, we read `this.state.count`:
+在 class 中我們要秀出目前的計數，我們使用 `this.state.count`：
```
Comment (neighborhood999) [inline content/docs/hooks-state.md]:
```suggestion
在 class 中我們要顯示目前的計數，我們使用 `this.state.count`：
```

File: content/docs/hooks-state.md [hunk 6]
```diff
@@ -212,69 +213,69 @@ Let's now **recap what we learned line by line** and check our understanding.
...
 function ExampleWithManyStates() {
-  // Declare multiple state variables!
+  // 宣告多個 state 變數！
   const [age, setAge] = useState(42);
   const [fruit, setFruit] = useState('banana');
   const [todos, setTodos] = useState([{ text: 'Learn Hooks' }]);
 ```
 
-In the above component, we have `age`, `fruit`, and `todos` as local variables, and we can update them individually:
+在上面的 component 中，我們的 local variable 有 `age`、`fruit` 和 `todos`，而且我們可以單獨更新它們。
```
Comment (neighborhood999) [inline content/docs/hooks-state.md]:
```suggestion
在上面的 component 中，我們的 local 變數有 `age`、`fruit` 和 `todos`，而且我們可以單獨更新它們。
```

File: content/docs/hooks-state.md [hunk 7]
```diff
@@ -212,69 +213,69 @@ Let's now **recap what we learned line by line** and check our understanding.
...
+## 下一步 {#next-steps}
 
-On this page we've learned about one of the Hooks provided by React, called `useState`. We're also sometimes going to refer to it as the "State Hook". It lets us add local state to React function components -- which we did for the first time ever!
+在這一頁，我們學習了其中一個 React 提供的 Hook－`useState`。我們有時也稱它做「State Hook」。前所未有地，它讓我們能夠增加 local state 到 React function component！
 
-We also learned a little bit more about what Hooks are. Hooks are functions that let you "hook into" React features from function components. Their names always start with `use`, and there are more Hooks we haven't seen yet.
+我們同時也瞭解了更多什麼是 Hook。Hook 是可以讓你在 function component 使用 React 各項功能的 function。還有更多 Hook 我們還沒認識，而 Hook 的名字都會以 `use` 起頭。
 
-**Now let's continue by [learning the next Hook: `useEffect`.](/docs/hooks-effect.html)** It lets you perform side effects in components, and is similar to lifecycle methods in classes.
+**現在我們繼續[學習下個 Hook：`useEffect`](/docs/hooks-effect.html)。** 就像 class 的 lifecycle method，它讓你能夠在 component 使用 side effect。
```
Comment (neighborhood999) [inline content/docs/hooks-state.md]:
```suggestion
**現在我們繼續[學習下個 Hook：`useEffect`](/docs/hooks-effect.html)。** 就像 class 的生命週期方法，它讓你能夠在 component 使用 side effect。
```

**Comment by piovischioh** [comment]
fixed and updated 😀

**Pull request merged**